File: src/1417-Reformat-The-String/1417.py
class Solution:
    def reformat(self, s: str) -> str:
        nums, chars = [], []
        for i in s:
            if '0' <= i <= '9': nums.append(i)
            else: chars.append(i)
                
        if abs(len(nums) - len(chars)) > 1: return ""
        res = ""
        flag = True if len(nums) >= len(chars) else False
        for _ in range(len(s)):
            if flag: res += nums.pop()
            else: res += chars.pop()
            flag = not flag
        return res

File: src/1417-Reformat-The-String/1417.cpp
class Solution {
public:
    string reformat(string s) {
        stringstream schars, snums;
        for (char c : s) {
            if (c >= '0' && c <= '9') snums << c;
            else schars << c;
        }
        
        string chars = schars.str(), nums = snums.str();
        if (abs(int(chars.size() - nums.size())) > 1) return "";
        stringstream res;
        
        bool flag = nums.size() >= chars.size() ? true : false;
        for (int i = 0; i < s.size(); i++) {
            if (flag) {
                res << nums.back();
                nums.pop_back();
            } else {
                res << chars.back();
                chars.pop_back();
            }
            flag = !flag;
        }
        return res.str();
    }
};